### PR TITLE
docs: replace the deprecated dropdownMatchSelectWidth

### PR DESCRIPTION
**Pull request by KurumiRin**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

Replace the deprecated dropdownMatchSelectWidth with popupMatchSelectWidth in the demo.
在demo中用popupMatchSelectWidth替换掉被废弃的dropdownMatchSelectWidth
`components/calendar/__tests__/index.test.tsx`一项测试用例的dropdownMatchSelectWidth也替换了一下

### 📝 Changelog

- Use a developer-oriented tone and narrative style.
- Describe the user's first-hand experience of the issue and its impact on developers, rather than your solution approach.
- Refer to: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### CODE REVIEW

**Review by crazyair — DISMISSED** [review]

**Comment by crazyair** [comment]
非组件部分，都改掉吧

**Review by afc163 — DISMISSED** [review]

**Comment by KurumiRin** [comment]
> 非组件部分，都改掉吧

改了config provider的，不过PropWarning里面的不确定要不要修改，方便看下不。剩余的都是测试警告废弃的，就不改了

**Review by crazyair — DISMISSED** [review]

File: components/config-provider/index.tsx [hunk 1]
```diff
@@ -548,7 +548,7 @@ const ProviderChildren: React.FC<ProviderChildrenProps> = (props) => {
 
   let childNode = (
     <>
-      <PropWarning dropdownMatchSelectWidth={dropdownMatchSelectWidth} />
+      <PropWarning popupMatchSelectWidth={popupMatchSelectWidth} />
```
Comment (crazyair) [inline components/config-provider/index.tsx:551]: 这个不应该改，不改组件代码。

Comment (KurumiRin) [reply components/config-provider/index.tsx:551]: 好的

**Review by crazyair — DISMISSED** [review]

**Comment by crazyair** [comment]
报错解决下

**Comment by crazyair** [comment]
应该是单测改了 api，快照要更新

**Comment by KurumiRin** [comment]
> 应该是单测改了 api，快照要更新

回退了一下config-provider，用例修改不符合测试结果，其他快照都更新了下。

**Review by crazyair — APPROVED** [review]

**Comment by crazyair** [comment]
> ## 👁 Visual Regression Report for PR #50027 Failed ❌
> > **🎯 Target branch:** master ([38cf5e8](https://github.com/ant-design/ant-design/commit/38cf5e8a75f83ca459e6535cd6c3458d8ae48712))
> > 📖 [View Full Report ↗︎](https://antd-visual-diff.oss-cn-shanghai.aliyuncs.com/pr-50027/visualRegressionReport/report.html)
> 
> Expected (Branch master)	Actual (Current PR)	Diff
>  table-summary.compact.png	 table-summary.compact.png	
> Check [Full Report](https://antd-visual-diff.oss-cn-shanghai.aliyuncs.com/pr-50027/visualRegressionReport/report.html) for details
> 
> If you think the visual diff is acceptable, please check:
> 
> * [ ]  Visual diff is acceptable

你看在你分支的这个 demo 对不对

**Comment by KurumiRin** [comment]
> > ## 👁 Visual Regression Report for PR #50027 Failed ❌
> > > **🎯 Target branch:** master ([38cf5e8](https://github.com/ant-design/ant-design/commit/38cf5e8a75f83ca459e6535cd6c3458d8ae48712))
> > > 📖 [View Full Report ↗︎](https://antd-visual-diff.oss-cn-shanghai.aliyuncs.com/pr-50027/visualRegressionReport/report.html)
> > 
> > 
> > Expected (Branch master)	Actual (Current PR)	Diff
> >  table-summary.compact.png	 table-summary.compact.png	
> > Check [Full Report](https://antd-visual-diff.oss-cn-shanghai.aliyuncs.com/pr-50027/visualRegressionReport/report.html) for details
> > If you think the visual diff is acceptable, please check:
> > 
> > * [ ]   Visual diff is acceptable
> 
> 你看在你分支的这个 demo 对不对

CleanShot 2024-07-24 at 14 14 17
正常的，看了下和官网文档表现一致

**Pull request merged**